src/import/pull-job.c

```diff
@@ -73,6 +75,33 @@ static void pull_job_finish(PullJob *j, int ret) {
                 j->on_finished(j);
 }
 
+static int pull_job_restart(PullJob *j) {
+        int r;
+        char *chksum_url = NULL;
+
+        r = import_url_change_last_component(j->url, "SHA256SUMS", &chksum_url);
+        if (r < 0)
+                return r;
+
+        free(j->url);
+        free(j->payload);
+        j->url = chksum_url;
+        j->state = PULL_JOB_INIT;
+        j->payload = NULL;
```

Comment:
not that it matters, but you can write this as:

```c
j->payload = mfree(j->payload);
```

and turn two seperate lines dealing with the same field into just one...